engine/src/test/java/org/camunda/bpm/engine/test/bpmn/event/timer/BoundaryTimerNonInterruptingEventTest.java

```diff
@@ -755,6 +762,54 @@ public void shouldExecuteTimeoutListenerJobOnOrAfterDueDate() {
     assertThat(timerJob.getDuedate()).isEqualTo(timerDueDate);
   }
 
+  @Test
+  @Deployment
+  public void shouldReevaluateTimerCycleWhenDue() throws Exception {
+    // given
+    MyBean myBean = new MyBean("R2/PT1H");
+    processEngineConfiguration.getBeans().put("myBean", myBean);
+    processEngineConfiguration.setReevaluateTimeCycleWhenDue(true);
+
+    runtimeService.startProcessInstanceByKey("nonInterruptingCycle").getId();
+
+    JobQuery jobQuery = managementService.createJobQuery();
+    assertThat(jobQuery.count()).isEqualTo(1);
+    moveByHours(1); // execute first job
+    assertThat(jobQuery.count()).isEqualTo(1);
+
+    // when bean changed and job is due
+    myBean.setCycle("R2/PT2H");
+    moveByHours(1); // execute second job
+
+    // then one more job is left due in 2 hours
+    assertThat(jobQuery.count()).isEqualTo(1);
+    assertThat(jobQuery.singleResult().getDuedate())
+        .isEqualToIgnoringMinutes(new Date(ClockUtil.getCurrentTime().getTime() + ((2 * 60 * 1000 * 60))));
+  }
+
+  @Test
+  @Deployment(resources = {"org/camunda/bpm/engine/test/bpmn/event/timer/BoundaryTimerNonInterruptingEventTest.shouldReevaluateTimerCycleWhenDue.bpmn20.xml"})
+  public void shouldNotReevaluateTimerCycleWhenDue() throws Exception {
```

Comment:
🔧 Suggestion: add a test case that does not re-evaluate the timer when the flag is enabled and the expression does not change. We test this for start timers if I am not mistaken, but not for other timers yet. Maybe it's also enough to have the functionality itself tested once in the start timers and only test that the flag generally works here for intermediate timers, no strong opinion here.

Reply:
I will add one test case, it's worth to have it in my opinion.